allow reusing Postgres scalars in custom types (close #4125) (#4333)
hasura/graphql-engine: docs/graphql/manual/actions/types/index.rst

These scalars can be used as arguments of the mutation or as fields of object
types and input types.

.. admonition:: Postgres scalars

>    Postgres base types are reusable as custom scalars without explicitly declaring them.

Reviewer: @rakeshkky do you think we need to explain a little more what this means? Like that the PG base types will be exposed as GraphQL scalars by Hasura? Since I don't understand this in depth, it's hard for me to make specific suggestions. Maybe @lexi-lambda can help?

Author: @marionschleifer I tried to improve the documentation in the latest commit.

Reviewer: Oh, didn't see that, sorry!

>    The base types from Postgres can be reusable as custom scalars without explicitly declaring them.

Reviewer: ```suggestion
   The Postgres base types are reusable as custom scalars without explicitly declaring them.
```

>    For example, the ``uuid`` type from ``pgcrypto`` extension can be used to resolve a UUID data.

Reviewer: ```suggestion
   For example, the ``uuid`` type from the ``pgcrypto`` extension can be used to resolve a ``uuid`` data field.
```

Reviewer: @rakeshkky I was not sure what you meant by "resolve a uuid data". I added a suggestion, please check for correctness 🙂
